editor/src/components/navigator/left-pane.tsx

```diff
@@ -1376,21 +1416,114 @@ const GithubPane = React.memo(() => {
   )
 })
 
-const FileChanges = ({ files, type }: { files: string[]; type: GithubFileStatus }) => {
-  if (files.length === 0) {
+const Ellipsise = ({
+  children,
+  title,
+  style,
+}: {
+  children: any
+  title?: string
+  style?: React.CSSProperties
+}) => {
+  return (
+    <div
+      style={{
+        ...style,
+        whiteSpace: 'nowrap',
+        overflow: 'hidden',
+        textOverflow: 'ellipsis',
+      }}
+      title={title}
+    >
+      {children}
+    </div>
+  )
+}
+
+const handleClickRevertAllFiles = (dispatch: EditorDispatch) => (e: React.MouseEvent) => {
+  e.preventDefault()
+  dispatch([EditorActions.showModal({ type: 'file-revert-all' })], 'everyone')
+}
+
+const handleClickRevertFile =
+  (item: GithubFileChangesListItem, dispatch: EditorDispatch) => (e: React.MouseEvent) => {
+    e.preventDefault()
+    dispatch(
+      [
+        EditorActions.showModal({
+          type: 'file-revert',
+          filePath: item.filename,
+          status: item.status,
+        }),
+      ],
+      'everyone',
+    )
+  }
+
+const RevertButton = ({
+  disabled,
+  text,
+  onMouseUp,
+}: {
+  disabled: boolean
+  text: string
+  onMouseUp: (e: React.MouseEvent) => void
+}) => {
+  return (
+    <Button
+      style={{ padding: '0 6px' }}
+      spotlight
+      highlight
+      disabled={disabled}
+      onMouseUp={onMouseUp}
+    >
+      {text}
+    </Button>
+  )
+}
+
+const GithubFileChangesList = ({
+  changes,
+  githubWorking,
+}: {
+  changes: GithubFileChanges | null
+  githubWorking: boolean
+}) => {
+  const count = React.useMemo(() => getGithubFileChangesCount(changes), [changes])
+  const dispatch = useEditorState((store) => store.dispatch, 'dispatch')
+  const list = React.useMemo(() => githubFileChangesToList(changes), [changes])
+
+  if (count === 0) {
     return null
   }
+
   return (
-    <div style={{ color: getGithubFileStatusColor(type), marginTop: 4, marginBottom: 4 }}>
-      <div>
-        {capitalize(type)} files ({files.length})
-      </div>
-      {files.map((f) => (
-        <div key={f} style={{ marginLeft: 8 }}>
-          &rarr; {f}
+    // Note: this is completely temporary until we finalize the design
```

Comment:
I don't think there's any need to include comments like this, just as long as we're tracking the UI work

Reply:
https://github.com/concrete-utopia/utopia/pull/2743/commits/b231d6f0df69151c4a2743c63b39233dcb459924